src/client/java/net/neoforged/neoforge/client/model/generators/blockstate/CompositeBlockStateModelBuilder.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.model.generators.blockstate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import net.minecraft.client.renderer.block.model.BlockStateModel;
+import net.minecraft.client.renderer.block.model.VariantMutator;
+import net.neoforged.neoforge.client.model.block.CompositeBlockModel;
+import org.jetbrains.annotations.Nullable;
+
+public class CompositeBlockStateModelBuilder extends CustomBlockStateModelBuilder {
+    @Nullable
+    private CompositeBlockModel.Unbaked model = null;
+
+    public void addPartModel(BlockStateModel.Unbaked partModel) {
+        List<BlockStateModel.Unbaked> models = new ArrayList<>();
```

Comment:
Isn't it more elegant to store a `List<BlockStateModel.Unbaked>` in the builder and turn it into a composite model at the end?